acra/acra-in-depth/security-design/_index.md

```diff
@@ -2,3 +2,165 @@
+
+Due to the deep integration of Acra into the application's data flow, Acra can detect suspicious activity in a growing list of places:
+- Connections to AcraConnector,
+- AcraServer listener,
+- Requests to AcraServer,
+- Database responses via AcraServer,
+- Data content coming from the AcraWriter-Database-AcraServer roundtrip.
+
+Currently, we've implemented the first few sample triggers to outline the methodology:
+- Attempts to extract whole tables with sensitive data by "poisoning" them with records that would never be selected unless SELECT * is used.
+- Inaccurate connection attempts to AcraConnector from the processes that don't match the expected conditions (if the users running AcraConnector / App itself are on the same VM / physical machine). Such expected conditions can be process names, name substrings, etc.
+
+We plan to include more triggers into the future releases.
+
+Once the trigger is triggered, an automatic reaction can be executed. You can choose to either:
+- notify a 3rd party monitoring / alarm system or
+- stop data decryption.
+
+### Guarantee 3: Add encryption context
+
+By differentiating the sensitive data via [Zones]({{< ref "acra/acra-in-depth/cryptography-and-key-management/#zones-INVALID" >}}), Acra users can build an additional layer of security:
+- Zone keys are used for encrypting the AcraStruct's RK section. In this case, AcraServer will only be able to pick the right key and decrypt the data by knowing the correct zone. This limits the leak scope for every private key, even if AcraServer is partially compromised.
+- Zone IDs are used as context when encrypting Secure Cell, providing additional protection.
+
+Zone data is supplied to an app via JSON (i.e. `{"id": "...", "public_key": "..."}`) from [HTTP api]({{< ref "acra/configuring-maintaining/general-configuration/acra-server.md#http-api-INVALID">}}) or [console utility]({{< ref "acra/security-controls/transport-security/acra-connector.md#acraconnector-and-acrawriter-INVALID" >}}).
+
+
+## Security model for Acra
+
+### Possible threats
+
+> Note: We recommend taking a look at the [architectural scheme of Acra]({{< ref "acra/acra-in-depth/data-flow/#-INVALID" >}}) before continuing to read.
+
+Acra at its core is a set of tools that allow safeguarding the security of a database (running PostgreSQL or MySQL) against the known widespread threats.
+
+The most dangerous current known security threats are:
+- Threats related to excess / abuse of access privileges.
+- Data leaks caused by mistakes in deploy, configuration settings, backup thefts, etc.
+- SQL injections.
+- Denial of service.
+- Vulnerabilities in the database protocol.
+- Weak audit or an absence thereof.
+- Operational system vulnerabilities.
+- Unsafe handling of cryptographic keys.
+
+### Conditions of secure work
+Acra can perform its protective functions properly and protect from the security threats 1, 2, 3 if the following security assumptions are met:
```

Comment:
> Conditions of secure work

“Preconditions” or “Basic assumptions” would be better, I think. The current wording reads somehow unnatural.

> the security threats 1, 2, 3

I guess this refers to the most common threats listed above, but the list is not numbered. It will be better to just name the threats here once again, it won't hurt.

Reply:
> “Preconditions” or “Basic assumptions” would be better, I think. The current wording reads somehow unnatural.

what difference between "conditionals" and "preconditionals" in this context? May be better use "requirements" instead of "conditions"... @storojs72 , what do you think?

> I guess this refers to the most common threats listed above, but the list is not numbered. It will be better to just name the threats here once again, it won't hurt.

we have numbered guarantees above and it refers to them, and they haven't been named. How you can name "compromisation systems except AcraServer" ?)

Reply:
Sorry for late review iteration. I agree with "Security assumptions" renaming of this small section. However I would expand it a bit by clarifying what "trusted" means for each item in list